### ISSUE TITLE
TreeGrid Sorting Should Leverage Bean's Comparable Implementation

### ISSUE THREAD

**Issue by bdunn44** [vaadin/framework#9447]
8.1.0.beta1

It's very difficult to control `TreeGrid` sorting when the current column-based configuration is too limited for the use case. I've had to override `createSortingComparator()` in order to sort my beans as I would expect: using the already-implemented `Comparable` interface.

```java
@Override
protected SerializableComparator<MyComparableBean> createSortingComparator() {
   return (o1, o2) -> o1.compareTo(o2);
}
```

**Comment by bdunn44** [comment]
Discovered `TreeDataProvider#setSortComparator()`. Closing.

**Comment by bdunn44** [comment]
Too fast. Setting the sort comparator for the data provider has no effect:
```java
TreeDataProvider<MyComparableBean> dataProvider = new TreeDataProvider<>(data);
dataProvider.setSortComparator((o1, o2) -> o1.compareTo(o2));
```
As of now the only way I've found to sort my items using their `Comparable` implementation is demonstrated in my first example. Reopening.

**Comment by Legioth** [comment]
Have you noticed the `Column.setComparator` method? You can use it to set an item comparator that is used whenever the component is configured to sort by that particular column (as long as in-memory data is used).

One reason for why `TreeDataProvider.setSortComparator` doesn't work for you could be that it's only supposed to be used as a tie breaker after any sorting applied through the component itself.

To further debug this issue, it might help if you set a breakpoint in `TreeDataProvider.fetchChildren` since that's the method that collects comparators from the component and the data provider itself and finally finds something to apply to the data using `Stream.sorted`.

**Comment by hesara** [comment]
I suspect that the reason setSortComparator() didn't work for you is the same internal issue that is behind #9449 - to confirm once it is addressed.

**Comment by hesara** [comment]
Probably addressed by #9449 in beta3 - please test once beta3 is released.
